### ISSUE TITLE
COCO dataset converter wrongly setup?

### ISSUE THREAD

**Issue by Data-drone** [mosaicml/streaming#567]
## Environment

- OS: Databricks Runtime 14.2 MLR - Ubuntu 22.04

## To reproduce

1. Download Coco 2017 dataset
2. Try to run the conversion script: https://github.com/mosaicml/streaming/blob/main/streaming/vision/convert/coco.py
3. Try to load the sharded data with StreamingCOCO: https://github.com/mosaicml/streaming/blob/main/streaming/vision/coco.py

This results in error: 

```
ValueError: Column bbox_labels contains an unsafe type: pkl. To proceed anyway, set ``allow_unsafe_types=True``.
```

Examining deeper it seems like maybe there is something with this line? 
https://github.com/mosaicml/streaming/blob/c503309d96bf4d52b14808c33ab6820f47e23113/streaming/vision/convert/coco.py#L117

Looking at the annotations json, the `image_id` field is a single integer, not a tuple? Is there a mistake in the parsing or are there different coco formats?

## Expected behavior

Everything should work and I should get the pkl error

**Comment by karan6181** [comment]
In the pickle dataset format, one can add any arbitrary code execution into the pickle data format and it's not secure for the system. Hence, we added a flag called `allow_unsafe_types` where if you are sure about the dataset, then provide `allow_unsafe_types=True`. The COCO dataset conversion script uses pickle data format for [bounding boxes](https://github.com/mosaicml/streaming/blob/main/streaming/vision/convert/coco.py#L205-L206). To unblock you, can you try passing `allow_unsafe_types=True` into `StreamingDataset` class?

**Comment by Data-drone** [comment]
Ah k the `StreamingCOCO` class can't pass through that flag?

**Comment by karan6181** [comment]
Yes, we might have missed that. If it works for you by providing `allow_unsafe_types=True`, do you mind creating a PR with a fix? Thanks!

**Comment by karan6181** [comment]
@Data-drone Wondering, did you got a chance to try out `allow_unsafe_types=True`?

**Comment by Data-drone** [comment]
I'll put in a pull soon just been caught up in other things

**Comment by snarayan21** [comment]
@Data-drone this has been addressed by #647, thanks for raising this!
